### PR TITLE
Use libc++ variant of string view and remove the FML variant.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: shell/common/shell.cc [hunk 1]
```diff
@@ -1114,11 +1111,11 @@ bool Shell::OnServiceProtocolRunInView(
...
 
   std::string main_script_path =
-      fml::paths::FromURI(params.at("mainScript").ToString());
+      fml::paths::FromURI(params.at("mainScript").data());
   std::string packages_path =
-      fml::paths::FromURI(params.at("packagesFile").ToString());
+      fml::paths::FromURI(params.at("packagesFile").data());
   std::string asset_directory_path =
-      fml::paths::FromURI(params.at("assetDirectory").ToString());
+      fml::paths::FromURI(params.at("assetDirectory").data());
```
Comment (gaaclarke) [inline shell/common/shell.cc]: Aren't these calls to .data() superfluous?  I believe a string_view casts to a string.

File: third_party/txt/tests/txt_test_utils.cc [hunk 2]
```diff
@@ -16,6 +16,8 @@
 
 #include "txt_test_utils.h"
 
+#include <sstream>
```
Comment (gaaclarke) [inline third_party/txt/tests/txt_test_utils.cc]: Erroneous include?

File: shell/common/shell.cc (hunk 1 above)
Comment (chinmaygarde) [reply shell/common/shell.cc]: Apparently not:
```
/Users/chinmaygarde/goma/gomacc  ../../buildtools/mac-x64/clang/bin/clang++ -MMD -MF obj/flutter/shell/common/common.shell.o.d -DUSE_OPENSSL=1 -DNO_TCMALLOC -DTOOLCHAIN_VERSION=bac220c15490dcf7b7d8136f75100bbc77e8d217 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -D_DEBUG -DSK_GL -DSK_ENABLE_DUMP_GPU -DSK_HAS_JPEG_LIBRARY -DSK_HAS_PNG_LIBRARY -DSK_HAS_WEBP_LIBRARY -DFLUTTER_RUNTIME_MODE_DEBUG=1 -DFLUTTER_RUNTIME_MODE_PROFILE=2 -DFLUTTER_RUNTIME_MODE_RELEASE=3 -DFLUTTER_RUNTIME_MODE=1 -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DUCHAR_TYPE=uint16_t -DSK_DISABLE_REDUCE_OPLIST_SPLITTING -DSK_LEGACY_SKCODEC_NONE_ENUM -DSK_GL -DSK_ENABLE_DUMP_GPU -DSK_DISABLE_AAA -DSK_DISABLE_READBUFFER -DSK_DISABLE_EFFECT_DESERIALIZATION -DSK_DISABLE_LEGACY_SHADERCONTEXT -DSK_DISABLE_LOWP_RASTER_PIPELINE -DSK_FORCE_RASTER_PIPELINE_BLITTER -DRAPIDJSON_HAS_STDSTRING -DRAPIDJSON_HAS_CXX11_RANGE_FOR -DRAPIDJSON_HAS_CXX11_RVALUE_REFS -DRAPIDJSON_HAS_CXX11_TYPETRAITS -DRAPIDJSON_HAS_CXX11_NOEXCEPT -I../.. -Igen -I../.. -I../../flutter/third_party/txt/src -I../../third_party/harfbuzz/src -I../../third_party/icu/source/common -I../../third_party/icu/source/i18n -I../../third_party/skia -I../../third_party/rapidjson -I../../third_party/rapidjson/include -I../../third_party -I../../third_party/dart/runtime -I../../third_party/dart/runtime/include -isysroot /Users/chinmaygarde/VersionControlled/MacOSX.sdk -mmacosx-version-min=10.12   -fno-strict-aliasing -fstack-protector-all -arch x86_64 -march=core2 -fcolor-diagnostics -Wall -Wextra -Wendif-labels -Werror -Wno-missing-field-initializers -Wno-unused-parameter -Wunguarded-availability -fvisibility=hidden -stdlib=libc++ -Wheader-hygiene -Wstring-conversion -Wthread-safety -O0 -g2 -fvisibility-inlines-hidden -std=c++17 -fno-rtti -fno-exceptions  -c ../../flutter/shell/common/shell.cc -o obj/flutter/shell/common/common.shell.o
../../flutter/shell/common/shell.cc:1113:54: error: no viable conversion from 'const std::__1::map<std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::less<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::allocator<std::__1::pair<const std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> > > > >::mapped_type' (aka 'const std::__1::basic_string_view<char, std::__1::char_traits<char> >') to 'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >')
  std::string main_script_path = fml::paths::FromURI(params.at("mainScript"));
                                                     ^~~~~~~~~~~~~~~~~~~~~~~
/Users/chinmaygarde/VersionControlled/engine/src/buildtools/mac-x64/clang/include/c++/v1/string:794:5: note: candidate constructor not viable: no known conversion from 'const std::__1::map<std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::less<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::allocator<std::__1::pair<const std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> > > > >::mapped_type' (aka 'const std::__1::basic_string_view<char, std::__1::char_traits<char> >') to 'const std::__1::basic_string<char> &' for 1st argument
    basic_string(const basic_string& __str);
    ^
/Users/chinmaygarde/VersionControlled/engine/src/buildtools/mac-x64/clang/include/c++/v1/string:799:5: note: candidate constructor not viable: no known conversion from 'const std::__1::map<std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::less<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::allocator<std::__1::pair<const std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> > > > >::mapped_type' (aka 'const std::__1::basic_string_view<char, std::__1::char_traits<char> >') to 'std::__1::basic_string<char> &&' for 1st argument
    basic_string(basic_string&& __str)
    ^
/Users/chinmaygarde/VersionControlled/engine/src/buildtools/mac-x64/clang/include/c++/v1/string:814:5: note: candidate constructor template not viable: no known conversion from 'const std::__1::map<std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::less<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::allocator<std::__1::pair<const std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> > > > >::mapped_type' (aka 'const std::__1::basic_string_view<char, std::__1::char_traits<char> >') to 'const char *' for 1st argument
    basic_string(const _CharT* __s) {
    ^
/Users/chinmaygarde/VersionControlled/engine/src/buildtools/mac-x64/clang/include/c++/v1/string:868:5: note: candidate constructor not viable: no known conversion from 'const std::__1::map<std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::less<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::allocator<std::__1::pair<const std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> > > > >::mapped_type' (aka 'const std::__1::basic_string_view<char, std::__1::char_traits<char> >') to 'initializer_list<char>' for 1st argument
    basic_string(initializer_list<_CharT> __il);
    ^
../../flutter/fml/paths.h:34:40: note: passing argument to parameter 'uri' here
std::string FromURI(const std::string& uri);
                                       ^

```

File: third_party/txt/tests/txt_test_utils.cc (hunk 2 above)
Comment (chinmaygarde) [reply third_party/txt/tests/txt_test_utils.cc]: Through a dependency chain, this was somehow being included earlier (probably `iosfwd` in the old string view). `std::stringstream` is used below so this is as close to IWYU as I could get.

**Pull request merged**